scripts/build.sh

```diff
@@ -20,12 +20,33 @@ elif [[ "$OSTYPE" == "darwin"* ]]; then
     CMAKE_FLAGS+="-DCMAKE_C_COMPILER=${XCODE_CMDLINE_DIR}/usr/bin/clang -DCMAKE_CXX_COMPILER=${XCODE_CMDLINE_DIR}/usr/bin/clang++ -DCMAKE_CXX_FLAGS=-isystem\ /usr/local/include -DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
 fi
 
-CMAKE_BUILD_TYPE="Debug"
-if [[ "$BUILD_RELEASE" == "1" ]]; then
+# Note:
+# CMAKE_BUILD_TYPE="Debug" adds "-O0 -g" flags by default
+# CMAKE_BUILD_TYPE="Release" adds "-O3 -DNDEBUG" by default
+if [[ "$BUILD_DEBUG" == "1" ]]; then
+    CMAKE_BUILD_TYPE="Debug"
+elif [[ "$BUILD_RELEASE" == "1" ]]; then
     CMAKE_BUILD_TYPE="Release"
 elif [[ "$BUILD_PROFILING" == "1" ]]; then
     CMAKE_BUILD_TYPE="Profiling"
 fi
 
+if [ $# -gt 0 ]; then
+    if [ "$1" == "Release" ]; then
```

Comment:
[optional] Thinking it could be helpful to have a 'help' arg for users using the cli and/or reading the script. This would show the options of Release|Profiling|Debug for `build.sh` if and only if the user enters 'help' or '--help' exactly.
```suggestion
    if [[ "$1" == "help" || "$1" == "--help" ]]; then
        echo "Usage: ./build.sh [Release|Profiling|Debug]"
        echo "If no argument is provided, the default build type is Debug."
        exit 0
    elif ...
```
Then would keep the existing 3 cases for Release, Profiling, Debug

Reply:
This is a good point, now that the script takes arguments, it makes sense to have a help flag. Will add this in

Reply:
Question for you here, do you think it makes sense to pass in arguments with dashes? I may switch to `--release` from `Release` to stick with our typical convention

Reply:
As a general rule, if there's going to be more than one option available to pass to a script, help output is a kindness. To that end, if the options aren't positional (i.e., you can pass more than one and in any order), they really should follow the `-l`/`--long` pattern.